apps/ios/SimpleX SE/ShareAPI.swift

```diff
@@ -76,6 +76,39 @@ func apiSendMessage(
             ttl: nil
         )
     )
-    if case let .newChatItem(_, chatItem) = r { return chatItem }
+    if case let .newChatItem(_, chatItem) = r {
+        return chatItem
+    } else {
+        if let filePath = cryptoFile?.filePath { removeFile(filePath) }
+        throw r
+    }
+}
+
+func apiActivateChat() throws {
+    chatReopenStore()
+    let r = sendSimpleXCmd(.apiActivateChat(restoreChat: false))
+    if case .cmdOk = r { return }
     throw r
 }
+
+func apiSuspendChat(expired: Bool) {
+    let r = sendSimpleXCmd(
+        .apiSuspendChat(
+            timeoutMicroseconds: expired
+            ? .zero
+            : 3_000_000
+        )
+    )
+    // Block until `chatSuspended` received or 3 seconds has passed
+    let startTime = CFAbsoluteTimeGetCurrent()
+    if case .cmdOk = r, !expired {
+        while CFAbsoluteTimeGetCurrent() - startTime < 3 {
+            switch recvSimpleXMsg() {
```

Comment:
```suggestion
            switch recvSimpleXMsg(messageTimeout: 3_000000) {
```

the idea here is that if it would fail to suspend within this time we can f